modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache

```diff
@@ -19,15 +19,23 @@ class URLSessionRequestBuilderFactory: RequestBuilderFactory {
     }
 }
 
+public typealias {{projectName}}APIChallengeHandler = ((URLSession, URLSessionTask, URLAuthenticationChallenge) -> (URLSession.AuthChallengeDisposition, URLCredential?))
+
 // Store the URLSession to retain its reference
-private var urlSessionStore = SynchronizedDictionary<String, URLSession>()
+private var urlSessionStore = SynchronizedDictionary<Int, URLSession>()
+
+// Store the URLSession's delegate to retain its reference
+private var sessionDelegateStore = SynchronizedDictionary<Int, SessionDelegate>()
```

Comment:
@DavidHorvath since we will have only one URLSession and one SessionDelegate, I think we can replace those two line with
```
 // Store the URLSession's delegate to retain its reference
private let sessionDelegate = SessionDelegate()

// Store the URLSession to retain its reference
private let defaultURLSession = URLSession(configuration: .default, delegate: sessionDelegate, delegateQueue: nil)
```